### DISCUSSION TITLE
Welcome to Structorizer Discussions!

### DISCUSSION THREAD

**Discussion by fesch** [fesch/Structorizer.Desktop#1106] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by Crane2feu** [comment]
Hello,
I found your application amazing.
But I have an idea for the application.
This idea is the add the "function" element for the includable routine.

**Reply by codemanyak** [reply]
@Crane2feu 
Thanks for your general appreciation. Considering your idea, however, I am afraid I don't quite comprehend it. Would you mind elaborating on it? What do you mean by "function" element? There is a function (subroutine) type of diagram, and there is a (subroutine) CALL element, but neither makes sense for me in this context (the former conceptually differs from an includable diagram, and the latter can be placed in an includable diagram easily). And what exactly does "includable routine" refer to? Could you provide some practical example, please?

**Reply by codemanyak** [reply]
@Crane2feu 
... or did you simply refer to the current awkward workaround of diverting disabled CALL elements to present method signatures in Includable diagrams serving as class representants on OOP language import? This might indeed make sense. This kind of element wouldn't be insertable via the GUI but could be used on import. I will consider it.

**Comment by Manoroc1** [comment]
I installed Java several times but structorizer still tells me it cant start without atleast java 11....

**Reply by fesch** [reply]
Did you tick the box that the JAVA_HOME environment variable will be set?

**Reply by Manoroc1** [reply]
yes i did

**Reply by codemanyak** [reply]
So then what OS are you working on (Windows, I guess)? What Java version have you installed? What does the console command `java -version` tell you?

**Reply by codemanyak** [reply]
You might also have a look at issue #1100 for more elaborate ways to find out what's going on. Don't hesitate to send us more information. This discussion tends to be rather an issue, by the way.

**Reply by Manoroc1** [reply]
java version "21.0.2" 2024-01-16 LTS
Java(TM) SE Runtime Environment (build 21.0.2+13-LTS-58)
Java HotSpot(TM) 64-Bit Server VM (build 21.0.2+13-LTS-58, mixed mode, sharing)

this it what it tells me

**Reply by codemanyak** [reply]
I derived a new issue #1124 from this thread. Please look there for the further steps and tests we may ask you for...

**Reply by codemanyak** [reply]
An also residing Java 1.8.something runtime environment (JRE) had interfered.
I just extended the respective [F.A.Q. entry](https://structorizer.fisch.lu/index.php?include=help) on the Structorizer [homepage](https://structorizer.fisch.lu) a bit in the hope that other users experiencing this frequent problem might find the clue right away.

**Comment by codemanyak** [comment]
## 🙏 
Hi folks, please consider ...
- starting a [new discussion](https://github.com/fesch/Structorizer.Desktop/discussions/new/choose) (rather than adding a comment to this one) if you are going to contribute new [ideas](https://github.com/fesch/Structorizer.Desktop/discussions/categories/ideas) or [threads](https://github.com/fesch/Structorizer.Desktop/discussions/categories/general) that do not actually go along with the headline of this general welcoming post or if you simply want to ask some [question](https://github.com/fesch/Structorizer.Desktop/discussions/categories/q-q);
- rather [generating an issue](https://github.com/fesch/Structorizer.Desktop/issues/new/choose) if you want to report some malfunction or bug or to request a specific new feature.

Thanks in advance 😃.
